### ISSUE TITLE
Access old and new collection values within CollectionChange type

### ISSUE THREAD

**Issue by ahimik** [javers/javers#1094]
Let's say we have a simple class with the Collection of primitives in it:

```
public class SimplePojo {
    private String name;
    private List<Integer> numbers;
}

```

the target goal is to compare two different objects of `SimplePojo` type and get the difference in the following output format:
```
property "name" has changed from "John" to "Sam".
property "numbers" has changed from "1, 2, 3" to "5, 6, 7".

```
,when I compare two objects using `javers.compare(obj1, obj2)` I receive two changes as expected: `ValueChange` for the `name` property and `ListChange` for the `numbers` property.
`ValueChange` has previous `getLeft()` and current `getRight()` values,
hovewer I couldn't find the way to access the previous and current collection values for the `ListChange` type.

Current workaround is to extract those values manually using reflection since `change.getAffectedObject()` provides a reference to the source object only.

The issue is similar to [this](https://github.com/javers/javers/issues/95) with the only difference that I need to access both values(previous and current) within a change.

**Describe the solution you'd like**
It would be nice to have a reference to the old and new values within any `CollectionChange` type similar to the `ValueChange` type.

**Comment by bartoszwalacik** [comment]
Hi, I agree that it could be useful. Contribution wanted :)

**Comment by bartoszwalacik** [comment]
usage:
```groovy
def "should provide old and new Collection in Changes loaded from Repository" () {
    given:
    def s = new SnapshotEntity(id:1, listOfIntegers : [1,2])
    javers.commit('author', s)
    
    s.listOfIntegers = [2,4]
    javers.commit('author', s)

    when:
    def changes = javers.findChanges(byInstanceId(1, SnapshotEntity).build())
    println(changes.prettyPrint())

    then:
    ListChange change = changes[0]
    change.left instanceof List
    change.left == [1,2]
    change.right instanceof List
    change.right == [2,4]
}
```

**Comment by bartoszwalacik** [comment]
released in 6.4.0
